### PR TITLE
Added county and city filter

### PR DESCRIPTION
**Pull request by lalalaurentiu**

### CODE REVIEW

**Review by sebiboga — APPROVED** [review]

File: src/components/search/search.component.jsx [hunk 1]
```diff
@@ -3,27 +3,62 @@ import "./search.style.scss";
...
+  const [countiesList, setCountiesList] = React.useState([counties_list]);
+  const [citiesList, setCitiesList] = React.useState([]);
+
+  React.useEffect(() => {
+    if (country === "România") {
+      setCountiesList(counties_list);
+    } else {
+      setCountiesList([]);
+    }
+  }, []);
```
Comment (eBitzu) [inline src/components/search/search.component.jsx:42]: dependency array not updated

File: src/components/search/search.component.jsx [hunk 2]
```diff
@@ -39,63 +74,98 @@ const handleClick = props.handleClick;
...
+    );
   };
 
+  // Update city search
   const updateCitySearch = (e) => {
     dispatch(updatCity(e.target.value));
-  }
 
+    counties.map((c) => {
+      if (Object.keys(c)[0] === county) {
```
Comment (eBitzu) [inline src/components/search/search.component.jsx:92]: can you please explain the logic here?

File: src/components/search/search.component.jsx [hunk 3]
```diff
@@ -39,63 +74,98 @@ const handleClick = props.handleClick;
...
+  // Handle click input
+  const handleClickInput = (e) => {
+    const dataList = e.target.nextElementSibling;
 
-      setCountiesList((countiesList) => [
-        ...countiesList,
-        Object.keys(county)[0],
-      ]);
-    
+    dataList.addEventListener("click", (d) => {
```
Comment (eBitzu) [inline src/components/search/search.component.jsx:118]: this is highly unoptimal

File: src/components/search/search.component.jsx [hunk 4]
```diff
@@ -39,63 +74,98 @@ const handleClick = props.handleClick;
...
+          if (e.target.value) {
+            setInputs(3);
+          }
+          break;
+        case "city":
+          dispatch(updatCity(e.target.value));
+          break;
+        default:
+          break;
+      }
```
Comment (eBitzu) [inline src/components/search/search.component.jsx:141]: where are you removing the event listener?

File: src/components/search/search.component.jsx (hunk 1 above)
Comment (lalalaurentiu) [reply src/components/search/search.component.jsx:42]: #351

File: src/components/search/search.component.jsx (hunk 2 above)
Comment (lalalaurentiu) [reply src/components/search/search.component.jsx:92]: This code updates the list of cities displayed based on user input for a specific county. It filters the cities to show only those that match the search criteria provided  by the user.

https://github.com/peviitor-ro/search-engine/assets/67306273/560a52ad-36d3-4062-9930-1c2bf2b80a0c

File: src/components/search/search.component.jsx (hunk 3 above)
Comment (lalalaurentiu) [reply src/components/search/search.component.jsx:118]: Why not?
Could you clarify it for me?

**Pull request merged**